### PR TITLE
Receipt 6 - Replace custom html receipts with Stripe's receipts

### PR DESCRIPTION
**Pull request by malinajirka**
Parent issue #4039 

Merge instructions
1. Make sure https://github.com/woocommerce/woocommerce-android/pull/4107 is merged
2. Remove "do not merge" label
3. Merge this PR

This PR removes all logic related to custom html receipts and replaces them with receipts generated by Stripe. Also adds support for emailing receipts.

I recommend review this PR by commits as otherwise the changes might look too big.

Known issues
- when the user clicks on "Print receipt" button it might take a few seconds before the Print UI gets shown -> we plan to introduce a progress indicator, but that's not part of this PR

To test:
1. Open detail of an unpaid order in US dollars
2. Tap "Collect payment" button
3. Walk through the connect + payment flows
4. Tap on "Send receipt" button and notice an email client with pre-filled content is shown
5. Go back to the app and tap on "Print receipt" button
6. Notice printing UI with Stripe's receipt is shown

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: libs/cardreader/src/debug/java/com/woocommerce/android/cardreader/internal/payments/PaymentManager.kt [hunk 1]
```diff
@@ -103,14 +101,18 @@ internal class PaymentManager(
...
-                emit(PaymentFailed(GENERIC_ERROR, null, e.message ?: "Unexpected error"))
-                return@flow
+            retrieveReceiptUrl(paymentIntent)?.let { receiptUrl ->
+                capturePayment(receiptUrl, orderId, cardReaderStore, paymentIntent)
             }
-            capturePayment(paymentInfo, orderId, cardReaderStore, paymentIntent)
+        }
+    }
+
+    private suspend fun FlowCollector<CardPaymentStatus>.retrieveReceiptUrl(
```
Comment (kidinov) [inline libs/cardreader/src/debug/java/com/woocommerce/android/cardreader/internal/payments/PaymentManager.kt:110]: This method looks but complicated to me. It returns a value and also it emits a value. Maybe just something like that would be easier to read:

```
        if (paymentIntent.status == PaymentIntentStatus.REQUIRES_CAPTURE) {
            val receiptUrl = paymentIntent.retrieveReceiptUrl()
            if (receiptUrl == null) {
                emit(PaymentFailed(GENERIC_ERROR, null, "ReceiptUrl not available"))
            } else {
                capturePayment(receiptUrl, orderId, cardReaderStore, paymentIntent)
            }
        }
    }

    private fun PaymentIntent.retrieveReceiptUrl() = getCharges().takeIf { it.isNotEmpty() }?.get(0)?.receiptUrl
```

?

File: WooCommerce/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -662,16 +662,9 @@
...
-    <string name="card_reader_receipt_application_name_title">Application name</string>
-    <string name="card_reader_receipt_from_title">Receipt from %s</string>
-    <string name="card_reader_receipt_title">Receipt</string>
-    <string name="card_reader_receipt_amount_paid_title">Amount paid</string>
-    <string name="card_reader_receipt_date_paid_title">Date paid</string>
-    <string name="card_reader_receipt_payment_method_title">Payment method</string>
-    <string name="card_reader_receipt_summary_title">Summary</string>
-    <string name="card_reader_receipt_aid_title">AID</string>
-
+`    <string name="card_reader_payment_receipt_email_subject">Receipt</string>
```
Comment (kidinov) [inline WooCommerce/src/main/res/values/strings.xml]: Maybe to put in a subject a bit more info, e.g. "Receipt for your purchase in X store"?

**Review by kidinov — APPROVED** [review]
Looks and works great! 👍

File: libs/cardreader/src/debug/java/com/woocommerce/android/cardreader/internal/payments/PaymentManager.kt (hunk 1 above)
Comment (malinajirka) [reply libs/cardreader/src/debug/java/com/woocommerce/android/cardreader/internal/payments/PaymentManager.kt:110]: I agree with you that returning and also emitting is not great, but I'm a bit on the edge here. The methods above this one - "collectPayment" and "processPayment" also emit + return, so I wanted to stay consistent. If we do what you propose with all of them, then `processPaymentIntent` becomes too complex. So I'm really not sure. Wdyt?

File: WooCommerce/src/main/res/values/strings.xml (hunk 2 above)
Comment (malinajirka) [reply WooCommerce/src/main/res/values/strings.xml]: Update in 9937e03de6a2a5f2f4dd67478b7a50bf80e5dcb4. I opted for a shorter version "Your receipt from xyz". Wdyt?

**Comment by malinajirka** [comment]
I'm merging this PR as the only change which hasn't been reviewed is the text of the subject. We can update it if needed in a separate PR.

File: libs/cardreader/src/debug/java/com/woocommerce/android/cardreader/internal/payments/PaymentManager.kt (hunk 1 above)
Comment (kidinov) [reply libs/cardreader/src/debug/java/com/woocommerce/android/cardreader/internal/payments/PaymentManager.kt:110]: Yeah, I see your point. Latter we could consider refactoring it a bit to make it more straightforward. Not a big deal for now

Comment (malinajirka) [reply libs/cardreader/src/debug/java/com/woocommerce/android/cardreader/internal/payments/PaymentManager.kt:110]: I'm all up for it ;) - having said that, I'm not sure how to make it more straight-forward. But if you have some ideas, go for it!

**Pull request merged**